### PR TITLE
Maybe 29 Refactor entities

### PR DESCRIPTION
**Pull request by niker000**
Refactored set field annotations in all entities

### CODE REVIEW

File: src/main/java/com/maybe/maybe/entity/enums/DeskState.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.maybe.maybe.entity.enums;
+
+public enum DeskState {
+    ALL,
+    AVAILABVLE
+}
```
Comment (ghost) [inline src/main/java/com/maybe/maybe/entity/enums/DeskState.java:6]: Where is constructor with enum IDs and its mapping?

File: src/main/java/com/maybe/maybe/entity/Desk.java [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+
+@Entity
+@Table(name = "desk")
+public class Desk extends AbstractEntity {
     @NotNull
-    @Column(name = "name",nullable = false)
+    @Column(name = "name", nullable = false)
     private String name;
 
     public String getName() {
```
Comment (ghost) [inline src/main/java/com/maybe/maybe/entity/Desk.java:16]: I think you have to add a field DeskState deskState in this entity and map it with enum DeskState

File: src/main/java/com/maybe/maybe/entity/Order.java [hunk 3]
```diff
@@ -49,12 +52,12 @@ public void setDateCreated(LocalDateTime dateCreated) {
         this.dateCreated = dateCreated;
     }
 
-    public Customer getCustomer() {
-        return customer;
+    public Desk getCustomer() {
```
Comment (ghost) [inline src/main/java/com/maybe/maybe/entity/Order.java]: Incorrect getter name

File: src/main/java/com/maybe/maybe/entity/Order.java [hunk 4]
```diff
@@ -49,12 +52,12 @@ public void setDateCreated(LocalDateTime dateCreated) {
...
+        return desk;
     }
 
-    public void setCustomer(Customer customer) {
-        this.customer = customer;
+    public void setCustomer(Desk desk) {
```
Comment (ghost) [inline src/main/java/com/maybe/maybe/entity/Order.java]: and incorrect setter name

File: src/main/java/com/maybe/maybe/entity/enums/DeskState.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.maybe.maybe.entity.enums;
+
+public enum DeskState {
+    ALL,
+    AVAILABVLE
```
Comment (ghost) [inline src/main/java/com/maybe/maybe/entity/enums/DeskState.java]: wrong value
must be AVAILABLE

**Review by ghost — COMMENTED** [review]
1. Enum DeskState mapping required
2. Some incorrect names

**Review by ghost — APPROVED** [review]

File: src/main/java/com/maybe/maybe/entity/ComponentProduct.java [hunk 6]
```diff
@@ -16,6 +17,7 @@
     @NotNull
     @ManyToOne
     @JoinColumn(name = "component_id", nullable = false)
+    @JsonManagedReference
```
Comment (CaH4aZzz) [inline src/main/java/com/maybe/maybe/entity/ComponentProduct.java]: Shouldn't we have here Collection of Component?

File: src/main/java/com/maybe/maybe/entity/enums/DeskState.java [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package com.maybe.maybe.entity.enums;
+
+public enum DeskState {
+    ALL,
```
Comment (deivur) [inline src/main/java/com/maybe/maybe/entity/enums/DeskState.java:4]: Dont understand status ALL? What it means?
I think there should be: AVAILABLE, NOT_AVAILABLE, ORDERED

**Review by CaH4aZzz — APPROVED** [review]

**Review by deivur — APPROVED** [review]

**Review by ArtemSobolenko — APPROVED** [review]

**Pull request merged**